### PR TITLE
Implement Initial DB Models

### LINKED ISSUE #22: Implement DB Schema
See #21

### PR DESCRIPTION
**Pull request by taitfuller**
Models have been implemented using Mongoose according to the design of #21. Implements an initial test suite for creating a valid Game.

Closes #22

### CODE REVIEW

**Review by RawiriHohepa — CHANGES_REQUESTED** [review]
Looks good for the most part, just a few small things with attributes, imports, and exports

File: backend/src/models/game.model.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  players?: Player[];
+  host?: ObjectId;
+  state?: GameState;
+  rounds?: Round[];
+}
+
+const GameSchema: Schema = new Schema({
+  gameCode: { type: String, required: true, unique: true },
+  settings: SettingsSchema,
+  setups: {
```
[the remark this answers is not in the document]
Comment (taitfuller) [reply backend/src/models/game.model.ts:32]: This is what the validate function achieves as required is not a valid property for Mongoose arrays 👌🏼

File: backend/src/models/index.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export { default as GameSchema, Game } from "./game.model";
+export { Player } from "./player.model";
```
[the remark this answers is not in the document]
Comment (taitfuller) [reply backend/src/models/index.ts:2]: Schema isn't actually required to create subdocuments so I'm only exporting the TypeScript types. Although I will rename `GameSchema` to `GameModel` to correctly reflect its type (we only have one actual model) and export GameState too (although we should probably refactor the enums `GameState` and `SetupType` to be common between the front and backend)

File: backend/src/models/game.model.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import mongoose, { Document, ObjectId, Schema } from "mongoose";
+import { SetupSchema, Setup } from "./setup.model";
```
[the remark this answers is not in the document]
Comment (taitfuller) [reply backend/src/models/game.model.ts:2]: Won't be exporting the Schema's from `index.ts` so will leave this

File: backend/src/models/round.model.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import { SetupSchema, Setup } from "./setup.model";
+
+export interface Round {
+  setup: Setup;
+  playersByPunchline: Map<string, ObjectId>;
+}
+
+export const RoundSchema: Schema = new Schema({
+  setup: { type: SetupSchema, required: true },
+  playersByPunchline: { type: Map, required: true },
```
[the remark this answers is not in the document]
Comment (taitfuller) [reply backend/src/models/round.model.ts:11]: Probably yes. At this point I have only implemented the initial functionality and tests to create a game for Milestone 1. Will be sure to test and fix this if necessary next week!

**Review by RawiriHohepa — APPROVED** [review]
Looks good, nice job!

**Pull request merged**